### DISCUSSION TITLE
How to make Renovate upgrade only to LTS versions?

### DISCUSSION THREAD

**Discussion by bzakdd** [renovatebot/renovate#31281] [Request Help] [answered]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Github / Renovate latest from dockerhub

### Please tell us more about your question or problem

How to make Renovate upgrade software to LTS versions only? I know that java has a specific preset:
https://docs.renovatebot.com/presets-workarounds/#workaroundsjavaltsversions
What about the other software?

**Comment by rarkins** [comment]
You need to be more specific. Naturally Renovate cannot know the LTS status of all software on the planet, but if there are distributions like Java, Node, Ubuntu etc which are widely used then we add awareness.

**Reply by bzakdd** [reply]
I am asking generally, because we want to apply this to many different types of software. I want to understand how can we support it with Renovate. There are some examples that I can solve:
Downloading latest LTS jenkins version van be done via customDatasource that finds LTS version from `https://get.jenkins.io/war-stable/`:
```json
"customDatasources": {
      "jenkins-lts": {
        "defaultRegistryUrlTemplate": "https://get.jenkins.io/war-stable/",
        "format": "html"
      }
    }
```
---

Finding the latest chrome version can be done by checking the searching through the debian `Packages` file:
```json
    "customDatasources": {
      "apt_google-chrome-stable": {
        "defaultRegistryUrlTemplate": "https://dl.google.com/linux/chrome/deb/dists/stable/main/binary-amd64/Packages",
        "format": "plain"
      }
    }
```
Then finding name of binary `google-chrome-stable` with `extractVersion`.

---

However in the both examples it works because they are marked by their maintainers as lts (in the first example) and stable (in the second example). However I don't know how to do it in, for example, case of Ansible:
https://docs.ansible.com/ansible/latest/reference_appendices/release_and_maintenance.html
Here I would need to implement logic that "if line `2.X` is the newest, then update to the newest patch under `2.(X-1)` line". Can I express it with Renovate?

What other tools does Renovate have to prevent updating to the newest version? My question is a followup of [this one](https://github.com/renovatebot/renovate/discussions/30527).

**Reply by rarkins** [reply, marked as answer]
If you insist on asking generically, then the answer is that there is no generic solution for LTS in the software industry and hence in Renovate either. LTS awareness needs to be added on a case by case basis, and where it's added then Renovate has reused its stable/unstable terminology to support it (e.g. not upgrading from stable to unstable) although this issue exists to make LTS awareness its own concept separate from stable/unstable: https://github.com/renovatebot/renovate/issues/27632
